compiler/rustc_hir_typeck/src/upvar.rs

```diff
@@ -2339,8 +2357,10 @@ fn determine_capture_info(
         }
     } else {
         // We select the CaptureKind which ranks higher based the following priority order:
-        // ByValue > MutBorrow > UniqueImmBorrow > ImmBorrow
+        // ByUse > ByValue > MutBorrow > UniqueImmBorrow > ImmBorrow
```

Comment:
Hmm, the ordering of `ByUse` and `ByValue` seems unclear to me, but I also expect that you never see them together? Maybe we should say `(ty::UpvarCapture::ByUse, ty::UpvarCapture::ByMove) => bug!` (and vice versa) instead. -- I'd probably change the code here to not use `_` but explicitly say `(ByValue, ByRef(_)) => lhs`, etc.